src/lib/core/CHIPConfig.h

```diff
@@ -2498,6 +2498,41 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
 #define CHIP_CONFIG_VERBOSE_VERIFY_OR_DIE 0
 #endif
 
+/**
+ * @def CHIP_CONFIG_MAX_GROUPS_PER_FABRIC
+ *
+ * @brief Defines the number of groups supported per fabric, see Group Key Management Cluster in specification.
+ *
+ * Binds to number of GroupState entries to support per fabric
+ */
+#ifndef CHIP_CONFIG_MAX_GROUPS_PER_FABRIC
+#define CHIP_CONFIG_MAX_GROUPS_PER_FABRIC 5
+#endif
+
+// TODO: Need to cap number of KeySets
+
+/**
+ * @def CHIP_CONFIG_MAX_GROUP_ENDPOINTS_PER_FABRIC
+ *
+ * @brief Defines the number of "endpoint->controlling group" mappings per fabric.
+ *
+ * Binds to number of GroupMapping entries per fabric
+ */
+#ifndef CHIP_CONFIG_MAX_GROUP_ENDPOINTS_PER_FABRIC
+#define CHIP_CONFIG_MAX_GROUP_ENDPOINTS_PER_FABRIC 8
+#endif
+
+/**
+ * @def CHIP_CONFIG_MAX_GROUP_CONCURRENT_ITERATORS
+ *
+ * @brief Defines the number of simultaneous Group iterators that can be allocated
+ *
+ * Number of iterator instances that can be allocated at any one time
+ */
+#ifndef CHIP_CONFIG_MAX_GROUP_CONCURRENT_ITERATORS
+#define CHIP_CONFIG_MAX_GROUP_CONCURRENT_ITERATORS 4
```

Comment:
I think 1 or 2 would suffice here. I don't foresee needing concurrent iteration outside of the main processing contexts of incoming messages, or outgoing messages. In any case, since there is no direct synchronization, we are reliant on external synchronization of the stack and therefore it's very unlikely more than 1 at a time would ever be at play.

Reply:
Default CHIP_CONFIG_MAX_GROUP_CONCURRENT_ITERATORS set to 2